cpp/src/parquet/arrow/arrow_schema_test.cc

```diff
@@ -999,6 +999,17 @@ TEST_F(TestConvertArrowSchema, ParquetLists) {
     auto arrow_list = ::arrow::list(arrow_element);
     arrow_fields.push_back(::arrow::field("my_list", arrow_list, false));
   }
+  // Same as above but using list_view as arrow type.
```

Comment:
The only thing that changes below is the list factory function, this could perhaps be factored out (also with `list_cast_fns` below), e.g.:
```c++
auto list_type_factory = [](const std::shared_ptr<::arrow::Field> field) {
    return ::arrow::list(field->type());
};

...

for (const auto& list_factory : {list_type_factory, list_view_type_factory}) { ... }
```